volatility3/framework/symbols/windows/extensions/shimcache.py

```diff
@@ -126,8 +132,12 @@ def file_path(self) -> Union[str, interfaces.renderers.BaseAbsentValue]:
             return self._file_path
 
         if not hasattr(self.Path, "Buffer"):
-            return self.Path.cast(
-                "string", max_length=self.Path.vol.count, encoding="utf-16le"
+            return address_to_string(
+                self._context,
+                self.Path.vol.layer_name,
+                self.Path.vol.offset,
+                520,
```

Comment:
Shouldn't this be `self.Path.size` or `self.Path.vol.count`?  Hard coding 520 just kinda sticks out as potentially allowing too much to be read (or forcing too little for future versions)...

Reply:
Good catch, fixed